core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintFactory.java

```diff
@@ -65,6 +67,14 @@ public interface ConstraintFactory {
      */
     <A> @NonNull UniConstraintStream<A> forEach(@NonNull Class<A> sourceClass);
 
+    /**
+     * As defined by {@link #forEach(Class)},
+     * but including filtered entities that match the supplied {@link ForEachInclude includes}.
```

Comment:
Should mention that the enum values imply an "AND" relationship.

Reply:
It an OR relationship, but the filter is negated, and by De Morgan's laws, the negation of an OR is an AND of their negations.
As for why the filter is negated, it because we take the complement set, and we need to include all assigned/consistent entities that would be rejected by an "only unassigned"/"only inconsistent" filter.